Sends request to create a folder.

webinista opened the pull request:
- Resets Create New Folder form when modal is closed.
- Triggering an error if the veteran submits an empty form without touching the input.
- Changing createNewFolder to createNewFolderSettings in config.
- Removing stray comments from reducers/messages.
- Making width value for .va-modal-close 'auto' (closes department-of-veterans-affairs/messaging-fe#79)

webinista committed:
Sends request to create a folder.

- Resets Create New Folder form when modal is closed.
- Triggering an error if the veteran submits an empty form without touching the input.
- Changing createNewFolder to createNewFolderSettings in config.
- Removing stray comments from reducers/messages.
- Making width value for .va-modal-close 'auto' (closes department-of-veterans-affairs/messaging-fe#79)

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#68

webinista commented:
May need to make some adjustments when department-of-veterans-affairs/vets-api/issues#168 is settled.

U-DON commented on src/js/messaging/containers/Main.jsx:
```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
-  handleSubmit() {}
+  handleCreateNewFolderModal() {
+    // Reset the form / state object when closed
+    const resetForm = { value: '', dirty: false };
```
Instead of creating the object ourselves, just pass `makeField('')` to `setnewFolderName`?

Also, alternatively, you can also have this within the handler for `TOGGLE_CREATE_FOLDER` so that it resets the state only if it's turning to not visible.

U-DON commented on src/js/messaging/actions/folders.js:
```diff
@@ -56,11 +57,21 @@ export function toggleManagedFolders() {
-    folderName
+  // create JSON payload
+  const folderData = {};
+  folderData.folder = folderName;
```
As explained by department-of-veterans-affairs/vets-api#168, we should be sending:

```
{
  folder: {
    name: folderName
  }
}
```

U-DON commented on src/js/messaging/containers/Main.jsx:
```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
+    }
+  }
+
+  validateFolderName(existingFolders = [], folderName, dirty = false) {
```
I think the validation should've stayed in the modal class. It would still work by passing the available folders as a prop to the modal.

U-DON replied:
Although syntactically still correct, I think arguments with default values should go after those that don't have one, so that something like `validateFolderName('foo')` won't end up assigning `existingFolders` to `'foo'`.

U-DON commented on src/js/messaging/reducers/modals.js:
```diff
@@ -16,7 +16,7 @@ const initialState = {
     visible: false
   },
   createFolder: {
-    visible: false,
```
Was this meant to be only for testing?

U-DON commented on src/js/messaging/containers/Main.jsx:
```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
+    this.props.toggleCreateFolderModal();
+  }
+
+  handleSubmit(domEvent) {
```
This should probably be renamed to `handleSubmitCreateFolder` or something now to be more specific.

U-DON commented on src/js/messaging/containers/Main.jsx:
```diff
@@ -35,13 +41,66 @@ class Main extends React.Component {
+    domEvent.preventDefault();
+    const input = domEvent.target.getElementsByTagName('input')[0];
+    // If, by chance, the veteran has submitted this form without touching the
+    // folder name field, trigger an action that will trigger an error.
```
Can we instead focus the field by default when the modal pops up?

webinista replied:
I tried that and couldn't figure out a way to do it since this modal is added and removed from the DOM when visible/not. Nothing I tried work. Maybe with `componentDidUpdate`? Can revisit later.

webinista committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#68

webinista committed:
Adds form handling for CreateNewFolder modal.

Closes department-of-veterans-affairs/messaging-fe#68

- Renamed handleSubmit method, reordered validateFolderName parameters.
- Moved form validation back to ModalCreateFolder.
- Handles create new folder response by appending response to data.items.
- Closes Create New Folder modal onSubmit.

Unblocks department-of-veterans-affairs/messaging-fe#82

webinista committed:
Merge branch 'department-of-veterans-affairs/messaging-fe#68' of github.com:department-of-veterans-affairs/vets-website into department-of-veterans-affairs/messaging-fe#68

webinista committed:
Setting modals.createFolder.visible back to false.

webinista committed:
Updating error text for createNewFolderMessages.exists.

U-DON committed:
Merge branch 'master' into department-of-veterans-affairs/messaging-fe#68

U-DON commented:
LGTM.
